Emotion 11 upgrade

RichardPK opened the pull request:
Resolves https://github.com/bbc/simorgh/issues/8348

**Overall change:** 

- Update all Psammead packages to use Emotion 11. The core migration involved the following changes within root psammead, and then in each package that had one of these dependencies:
```json
    "@emotion/core": "^10.0.35"  >  "@emotion/react": "^11.1.4"
    "@emotion/styled": "^10.0.27"  >  "@emotion/styled": "^11.0.0"
    "jest-emotion": "^10.0.32"  >  "@emotion/jest": "^11.1.0"
    "babel-plugin-emotion": "^10.0.33"  >  "@emotion/babel-plugin": "^11.1.2"
```

- These package updates triggered classNames to be regenerated for snapshots across all packages, and some rearranging of styles. The line count of all snapshots before/after migration is consistent though, with two exceptions:

Two packages had line changes in their snapshots - `psammead-radio-schedule` & `psammead-media-player/Placeholder`.  
- `psammead-media-player/Placeholder` had some janky styles, which were [fixed in this PR](https://github.com/bbc/psammead/pull/4193/files#diff-4c3929e4357421334316b48a45933497fa55ace8cd5e3d16d2a12a6b4d06ae3fR20) and restored snapshot parity
- `psammead-radio-schedule` has lost some snapshot lines but as detailed in Ryan's comments below, we think these are optimisations rather than the loss of styles.

**Notes**
- We removed `psammead-styles` from this singular upgrade PR due to widespread changes in its snapshots.
- `@emotion/core": "^10.1.1"` is also kept as a dependency while we still need to support Emotion 10 for psammead-styles


---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

RichardPK committed:
First commit

ryanmccombe commented on packages/components/psammead-media-player/src/Placeholder/__snapshots__/index.test.jsx.snap:
```diff
@@ -10,6 +10,16 @@ exports[`Media Player: Placeholder should render a video placeholder 1`] = `
   cursor: pointer;
 }
 
+.emotion-1:hover .css-9vh530-StyledPlayButton,
+.emotion-1:focus .css-9vh530-StyledPlayButton {
+  background-color: #B80000;
+}
+
```
These are unused duplicates of the styles below

Accounts for 5/40 of the new lines

RichardPK replied:
Closing all the placeholder snapshot comments - a fix to the styles achieved snapshot parity 👌

ryanmccombe commented on packages/components/psammead-media-player/src/Placeholder/__snapshots__/index.test.jsx.snap:
```diff
@@ -170,13 +175,68 @@ exports[`Media Player: Placeholder should render a video placeholder with guidan
   cursor: pointer;
 }
 
+.emotion-1:hover .css-9vh530-StyledPlayButton,
+.emotion-1:focus .css-9vh530-StyledPlayButton {
+  background-color: #B80000;
+}
```
These are unused duplicates of the styles below

Accounts for 5/40 of the new lines

RichardPK replied:
Closing all the placeholder snapshot comments - a fix to the styles achieved snapshot parity 👌

ryanmccombe commented on packages/components/psammead-media-player/src/Placeholder/__snapshots__/index.test.jsx.snap:
```diff
@@ -362,6 +372,16 @@ exports[`Media Player: Placeholder should render a video placeholder without dur
   cursor: pointer;
 }
 
+.emotion-1:hover .css-9vh530-StyledPlayButton,
+.emotion-1:focus .css-9vh530-StyledPlayButton {
+  background-color: #B80000;
+}
```
These are unused duplicates of the styles below

Accounts for 5/40 of the new lines

RichardPK replied:
Closing all the placeholder snapshot comments - a fix to the styles achieved snapshot parity 👌

ryanmccombe commented on packages/components/psammead-media-player/src/Placeholder/__snapshots__/index.test.jsx.snap:
```diff
@@ -509,6 +524,16 @@ exports[`Media Player: Placeholder should render an audio placeholder 1`] = `
   cursor: pointer;
 }
 
+.emotion-1:hover .css-9vh530-StyledPlayButton,
+.emotion-1:focus .css-9vh530-StyledPlayButton {
+  background-color: #B80000;
+}
```
These are unused duplicates of the styles below

Accounts for 5/40 of the new lines

RichardPK replied:
Closing all the placeholder snapshot comments - a fix to the styles achieved snapshot parity 👌

ryanmccombe commented on packages/components/psammead-media-player/src/Placeholder/__snapshots__/index.test.jsx.snap:
```diff
@@ -672,6 +692,16 @@ exports[`Media Player: Placeholder should render an audio placeholder without du
   cursor: pointer;
 }
 
+.emotion-1:hover .css-9vh530-StyledPlayButton,
+.emotion-1:focus .css-9vh530-StyledPlayButton {
+  background-color: #B80000;
+}
```
These are unused duplicates of the styles below

Accounts for 5/40 of the new lines

RichardPK replied:
Closing all the placeholder snapshot comments - a fix to the styles achieved snapshot parity 👌

ryanmccombe commented on packages/components/psammead-media-player/src/Placeholder/__snapshots__/index.test.jsx.snap:
```diff
@@ -805,83 +830,44 @@ exports[`Media Player: Placeholder should render an audio placeholder without du
-@media screen and (-ms-high-contrast:active) {
-  .emotion-0 {
-    background-color: window;
-  }
+.emotion-1:hover .css-1uhr2ah-Button-StyledPlayButton,
+.emotion-1:focus .css-1uhr2ah-Button-StyledPlayButton {
+  background-color: #B80000;
 }
 
-@media (min-width:25rem) {
-  .emotion-0 {
-    padding: 1rem;
-  }
+.emotion-1:hover .css-pz3gmj-StyledPlayButton,
+.emotion-1:focus .css-pz3gmj-StyledPlayButton {
+  background-color: #B80000;
 }
```
These are unused quadruplicates of the styles below

Accounts for 15/40 of the new lines

RichardPK replied:
Closing all the placeholder snapshot comments - a fix to the styles achieved snapshot parity 👌

ryanmccombe commented on packages/components/psammead-radio-schedule/src/__snapshots__/index.test.jsx.snap:
```diff
@@ -160,67 +150,57 @@ exports[`RadioSchedule should render ltr radio schedules correctly 1`] = `
 
-@media (max-width:14.9375rem) {
-    .emotion-41 {
-      width: initial;
-      margin: 0;
```
Emotion seems to be optimising this for us?  These two lines appear 6 times (once in every media query) in the "before"

In the after, they are just specified once, outside any media query

Accounts for 10 of the 40 removed lines

ryanmccombe commented on packages/components/psammead-radio-schedule/src/__snapshots__/index.test.jsx.snap:
```diff
@@ -1,112 +1,102 @@
 
-@media (max-width:14.9375rem) {
-    .emotion-161 {
-      width: initial;
-      margin: 0;
```
Emotion seems to be optimising this for us? These two lines appear 6 times (once in every media query) in the "before"

In the after, they are just specified once, outside any media query

Accounts for 10 of the 40 removed lines

ryanmccombe commented on packages/components/psammead-radio-schedule/src/__snapshots__/index.test.jsx.snap:
```diff
@@ -1233,112 +1213,102 @@ exports[`RadioSchedule should render ltr radio schedules correctly 1`] = `
 
-@media (max-width:14.9375rem) {
-    .emotion-159 {
-      width: initial;
-      margin: 0;
```
More optimisation - 12 lines removed, 2 added

ryanmccombe commented on packages/components/psammead-radio-schedule/src/__snapshots__/index.test.jsx.snap:
```diff
@@ -1392,67 +1362,57 @@ exports[`RadioSchedule should render rtl radio schedules correctly 1`] = `
 
-@media (max-width:14.9375rem) {
-    .emotion-39 {
-      width: initial;
-      margin: 0;
```
Optimised (-12 lines, +2 lines)

ryanmccombe committed:
Prep to fix merge conflicts

# Conflicts:
#	packages/components/psammead-most-read/src/Item/__snapshots__/index.test.jsx.snap
#	packages/components/psammead-most-read/src/List/__snapshots__/index.test.jsx.snap
#	packages/components/psammead-radio-schedule/package-lock.json
#	packages/components/psammead-radio-schedule/src/__snapshots__/index.test.jsx.snap

ryanmccombe committed:
Fix conflicts

Commit:
Update chromaticqa action

Commit:
Set git fetch dept to 0 to fetch all git commit history

RichardPK committed:
Improvements to media player placeholder styles in order to fix the snapshots

RichardPK committed:
Fix merge conflicts

RichardPK committed:
Regen package lock after merge from latest

ryanmccombe committed:
Fix conflicts

# Conflicts:
#	packages/components/psammead-media-player/package-lock.json
#	packages/components/psammead-media-player/package.json

ryanmccombe committed:
Patch bumps - tranche 1

ryanmccombe committed:
Patch bumps - tranche 2

ryanmccombe committed:
Patch bumps - tranche 3

ryanmccombe committed:
Patch bumps - tranche 4

ryanmccombe committed:
Patch bumps - tranche 5

ryanmccombe committed:
Patch bumps - tranche 6

ryanmccombe committed:
Patch bump - tranche 7

ryanmccombe committed:
Bump psammead-assets

ryanmccombe committed:
Merge branch 'latest' into emotion-11-upgrade

ryanmccombe committed:
Patch bump - psammead-locales

ryanmccombe committed:
Merge branch 'emotion-11-upgrade' of github.com:bbc/psammead into emotion-11-upgrade

ryanmccombe committed:
Patch bump - psammead root

ryanmccombe committed:
Fix podcast-promo merge conflicts

ryanmccombe committed:
Fix snapshots

RichardPK commented on .github/workflows/psammead-chromaticqa.yml:
```diff
@@ -17,10 +17,11 @@ jobs:
         node-version: ['12']
     env:
       CI: true
-      CHROMATIC_APP_CODE: ${{ secrets.CHROMATIC_APP_CODE }}
```
Are these changes expected @ryanmccombe?

ryanmccombe replied:
aye - Chris added these changes to fix an issue with Chromatic failing on CI

RichardPK committed:
Merge branch 'latest' into emotion-11-upgrade

ryanmccombe committed:
Fix conflicts

rebeccamcginn reviewed (approved):
Go for it

RichardPK committed:
Fixing merge conflicts after psammead-styles cascading updates

RichardPK committed:
Rebuilding package-lock to fix merge conflicts

RichardPK committed:
Fix root changelog.md

RichardPK committed:
Bump psammead-byline
